### PR TITLE
Fix getColumns column mapping 

### PR DESCRIPTION
**Pull request by grantatspothero**
There is not a great way to test the missing `fromRemoteColumnName` call since right now the call does nothing besides lowercase the column names (in `DefaultIdentifierMapping`). However, other parts of trino already lowercase column names which hides the bug. When adding the column mapping functionality that is when I found the bug: https://github.com/trinodb/trino/pull/9986

Also migrate memsql tests to basecaseinsensitive mappings as described in this issue: https://github.com/trinodb/trino/issues/7864

The tests I removed were copied 1<->1 from the basecaseinsensitivemappings tests with very minor changes (for example: using `l m u` as column values instead of `lower mixed upper`).

### CODE REVIEW

**Comment by grantatspothero** [comment]
Will reopen after fixing tests

**Comment by ebyhr** [comment]
Note that PR from forked branch won't run MemSQL tests. Leaving this comment for reminder to reviewers.

**Review by hashhar — COMMENTED** [review]
Maybe reword first commit to `Use remote column names when looking up timestamp column's precision in MemSQL`?

**Review by hashhar — COMMENTED** [review]
Maybe it would be better to have a single commit that fixes the bug in all clients + a second commit that migrates all affected to use BaseCaseInsenstiveMappingTests.

The impl seems good to me except that we need to take `DatabaseMetadata#storesUppercaseIdentifiers` into consideration in IdentifierMapping#fromRemoteColumnName.

**Review by hashhar — COMMENTED** [review]
Ah, seems the fix isn't straightforward. The `getColumns` is used both to talk to remote system (by the query builder classes) and internally (to create table schema and table metadata).

Internal usages don't run into a bug because they create `ColumnMetadata` from `ColumnHandle`. ColumnMetadata's constructor unconditionally lowercases the column name. That way all internal usages (as long as they use `ColumnMetadata` always get lowercased names) and external usages get actual names.

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java [hunk 1]
```diff
@@ -367,20 +367,21 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
                 List<JdbcColumnHandle> columns = new ArrayList<>();
                 while (resultSet.next()) {
                     allColumns++;
-                    String columnName = resultSet.getString("COLUMN_NAME");
+                    String remoteColumnName = resultSet.getString("COLUMN_NAME");
```
Comment (kokosing) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java:370]: Would it be possible to write some test that would be expose this bug?
